### PR TITLE
Feature/6088 use presence tracing config parameters to parse qr codes

### PR DESCRIPTION
**Pull request by 30mar**
## Description
We need to update our verification for QRcodes using regex according to the latest changes from the techspecs:
https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/event-registration-mvp/docs/spec/event-registration-client.md#attendee-check-in

**Approach**: I needed to move the whole verification to a helper class and make this class responsible for the creation of the trace location or returning an error
This class need to be accessible from the **checkinCoordinator** to be reused before showing the TraceLocationDetailVC
it is used twice:

1. if we scan a qr code internally and had no errors with the camera detection, we call it using the qrCodeString
2. using the native camera "Need to merged" with the same approach

- If no Errors happen we show the **TraceLocationDetailVC**
- If an Errors happen then we show it from the root view controller inside the coordinator "AKA **checkinableViewController**"

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6088

## Screenshots
https://user-images.githubusercontent.com/15270737/113046220-1949e480-91a0-11eb-8486-18391be49a55.mp4

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+				  let payLoadRange = Range(unWrappedMatch.range(at: Int(payLoadIndex) + 1), in: url)
+			else {
+				Log.error("the QRCode matched none of the regular expressions", log: .checkin)
+				onError(QRScannerError.codeNotFound)
+				return
+			}
+
+			// let version = url[versionRange]
+			let payLoad = url[payLoadRange]
+			let encodingType = EncodingType(rawValue: qrDescriptor.payloadEncoding.rawValue) ?? .unspecified
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift:55]: Why still this massive and error prone overhead?
```
let components = URL(string: url)!.pathComponents // shortcut version
let payload = components.popLast()
let encoding = components.popLast()
```

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/SAPDownloadedPackage.swift [hunk 2]
```diff
@@ -47,29 +47,7 @@ struct SAPDownloadedPackage: Fingerprinting {
...
-					continue
-				}
-
-				if publicKey.isValid(signature: signature, for: package.bin) {
-					return true
-				}
-			}
-
-			return false
+			return true
```
[the remark this answers is not in the document]
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/SAPDownloadedPackage.swift]: forgot it too, i used this for testing with mock server, just reverted

File: src/xcode/ENA/ENA/Source/RootCoordinator.swift [hunk 3]
```diff
@@ -119,24 +119,13 @@ class RootCoordinator: RequiresAppDependencies {
...
 			  let index = tabBarController.viewControllers?.firstIndex(of: checkInNavigationController) else {
 			return
 		}
 		tabBarController.selectedIndex = index
-		
-//		DispatchQueue.main.asyncAfter(wallDeadline: .now() + 0.5) {
-//
-//		}
-		checkInCoordinator.showTraceLocationDetails(traceLocation)
+		checkInCoordinator.showTraceLocationDetails(guid)
```
[the remark this answers is not in the document]
Comment (30mar) [reply src/xcode/ENA/ENA/Source/RootCoordinator.swift:128]: I'm not sure about the dispatch, but for the rest: the Router code is outdated and it is happening now inside the regex stuff, so all you need is to call the showCheckin public function and pass the qrCodeString that you have here, all checks will happen in another helper class. I did this so we don't duplicate the code

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift:55]: There is no guarantee. the URL class will stay this way , i,e need only the last 2 components, this is why they are sending these indices from the server to be able to select the correct group which can be more than 2 according to max, lets have a call with him tomorrow to discuss it but i will vote for your solution as well if he agrees, it is much simpler and to the point.

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/EncodingType.swift [hunk 4]
```diff
@@ -0,0 +1,11 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+enum EncodingType: Int {
```
[the remark this answers is not in the document]
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/EncodingType.swift:8]: I would even go so far and call it 
```suggestion
enum QRCodeDataEncodingType: Int {
```

**Comment by marcussc** [comment]
Really appreciate your detailed Description, gives a nice insight what is going on.

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewModel.swift [hunk 5]
```diff
@@ -4,13 +4,13 @@
...
 import Foundation
 import AVFoundation
-import OpenCombine
 
 final class CheckinQRCodeScannerViewModel: NSObject, AVCaptureMetadataOutputObjectsDelegate {
 
 	// MARK: - Init
 
-	override init() {
+	init(appConfiguration: AppConfigurationProviding) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewModel.swift:12]: Do you really need appConfiguration here?

**Review by freshking — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**